fe/src/main/java/org/apache/doris/load/BrokerFileGroupAggInfo.java

```diff
@@ -0,0 +1,239 @@
+                return true;
+            }
+            if (!(obj instanceof FileGroupAggKey)) {
+                return false;
+            }
+
+            FileGroupAggKey other = (FileGroupAggKey) obj;
+            return other.tableId == this.tableId && other.partitionIds.equals(this.partitionIds);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(tableId, partitionIds);
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("[").append(tableId).append(": ").append(partitionIds).append("]");
+            return sb.toString();
+        }
+    }
+
+    public BrokerFileGroupAggInfo() {
+
+    }
+
+    public void addFileGroup(BrokerFileGroup fileGroup) throws DdlException {
+        FileGroupAggKey fileGroupAggKey = new FileGroupAggKey(fileGroup.getTableId(), fileGroup.getPartitionIds());
+        List<BrokerFileGroup> fileGroupList = aggKeyToFileGroups.get(fileGroupAggKey);
+        if (fileGroupList == null) {
+            // check if there are overlapping partitions of same table
+            if (tableIdToPartitioIds.containsKey(fileGroup.getTableId()) 
+                    && tableIdToPartitioIds.get(fileGroup.getTableId()).stream().anyMatch(id -> fileGroup.getPartitionIds().contains(id))) {
+                throw new DdlException("There are overlapping partitions of same table in data descrition of load job stmt");
+            }
+            
+            fileGroupList = Lists.newArrayList();
+            aggKeyToFileGroups.put(fileGroupAggKey, fileGroupList);
+        }
+        // exist, aggregate them
+        fileGroupList.add(fileGroup);
+
+        // update tableIdToPartitioIds
+        Set<Long> partitionIds = tableIdToPartitioIds.get(fileGroup.getTableId());
+        if (partitionIds == null) {
+            partitionIds = Sets.newHashSet();
+            tableIdToPartitioIds.put(fileGroup.getTableId(), partitionIds);
+        }
+        if (fileGroup.getPartitionIds() != null) {
+            partitionIds.addAll(fileGroup.getPartitionIds());
+        }
+    }
+
+    public Set<Long> getAllTableIds() {
+        return aggKeyToFileGroups.keySet().stream().map(k -> k.tableId).collect(Collectors.toSet());
+    }
+
+    public Map<FileGroupAggKey, List<BrokerFileGroup>> getAggKeyToFileGroups() {
+        return aggKeyToFileGroups;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(aggKeyToFileGroups);
+        return sb.toString();
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        // The pull load source info doesn't need to be persisted.
+        // It will be recreated by origin stmt in prepare of load job.
+        // write 0 just for compatibility
+        out.writeInt(0);
+    }
+
+    public void readFields(DataInput in) throws IOException {
+        in.readInt();
```

Comment:
Maybe in is a PullLoadSourceInfo??

Reply:
It won't, we already solve this in version 0.11